ERCS/erc-7683.md

```diff
@@ -98,79 +121,253 @@ struct Input {
 struct Output {
 	/// @dev The address of the ERC20 token on the destination chain
 	/// @dev address(0) used as a sentinel for the native token
-	address token;
+	bytes32 token;
 	/// @dev The amount of the token to be sent
 	uint256 amount;
 	/// @dev The address to receive the output tokens
-	address recipient;
+	bytes32 recipient;
 	/// @dev The destination chain for this output
-	uint32 chainId;
+	uint64 chainId;
+}
+
+/// @title FillInstruction type
+/// @notice Instructions to parameterize each leg of the fill
+/// @dev Provides all the origin-generated information required to produce a valid fill leg
+struct FillInstruction {
+	/// @dev The contract address that the order is meant to be settled by
+	uint64 destinationChainId;
+	/// @dev The contract address that the order is meant to be filled on
+	bytes32 destinationSettler;
+	/// @dev The data generated on the origin chain needed by the destinationSettler to process the fill
+	bytes originData;
 }
+```
+
+### Open event
+
+A compliant `Open` event MUST adhere to the following abi:
 
+```solidity
+/// @notice Signals that an order has been opened
+/// @param orderId a unique order identifier within this settlement system
+/// @param fillInstructions instructions to parameterize each fill leg, with each element representing a single leg
+event Open(bytes32 indexed orderId, ResolvedCrossChainOrder resolvedOrder);
 ```
 
-### ISettlementContract interface
+### Settlement interfaces
 
-A compliant settlement contract implementation MUST implement the `ISettlementContract` interface:
+A compliant origin settler contract implementation MUST implement the `IOriginSettler` interface:
 
 ```solidity
-/// @title ISettlementContract
-/// @notice Standard interface for settlement contracts
-interface ISettlementContract {
-	/// @notice Initiates the settlement of a cross-chain order
-	/// @dev To be called by the filler
-	/// @param order The CrossChainOrder definition
-	/// @param signature The swapper's signature over the order
+/// @title IOriginSettler
+/// @notice Standard interface for settlement contracts on the origin chain
+interface IOriginSettler {
+	/// @notice Opens a gasless cross-chain order on behalf of a user.
+	/// @dev To be called by the filler.
+	/// @dev This method must emit the Open event
+	/// @param order The GaslessCrossChainOrder definition
+	/// @param signature The user's signature over the order
+	/// @param fillerData Any filler-defined data required by the settler
+	/// @returns fillInstructions instructions to parameterize each fill leg, with each element representing a single leg
+	function openFor(GaslessCrossChainOrder order, bytes signature, bytes originFillerData) external;
```

Comment:
How do fillers know how to populate this?

Reply:
It's a great question. Since, fundamentally, it's meant to be arbitrary protocol-specific information populated by the filler, there is no standard way to tell fillers _how_ to populate it.

All we can do is provide a place for the struct types to be shared: https://github.com/ERC-7683/subtypes/tree/main/filler-data-subtypes. 

Thoughts?